### DISCUSSION TITLE
Skip key when value is zero/empty

### DISCUSSION THREAD

**Discussion by dwaydwaydway** [stephenberry/glaze#1008] [Q&A]
Hi, 

Is there a way to skip key/value which is empty when writing json?
For example, 
```
struct A {
    double d = 0.0;
    std::string s{""};
    std::string a{"something"};
};

std::string result = glz::write_json(s);
```
I'm getting `{"d":0.00000005046102E-309, "s": "", "a": "something" }` when what I wanted is `{"a": "something" }`

I tried to play with the lambda function when doing value mapping but it didn't really work.
```
template <>
struct glz::meta<A> {
   static constexpr auto value = object(
      "s", [](auto&& self) -> auto& { return self.s.enpty() ? nullptr:self.s; } // compile error
   );
};
```

**Comment by stephenberry** [comment]
Currently the solution is to wrap these types in `std::optional`, which will skip writing them out if they are in the null optional state. However, I do plan to add support for what you want soon, it is an active issue here: #855 
My first priority is to implement some significant performance improvements, and then I'll start implementing more features like this again.
